### PR TITLE
[clang][OpenMP] Place some common code in functions

### PR DESCRIPTION
**Pull request by kparzysz**
There are chunks of code repeated in a number of functions. This patch moves some of that code into individual functions.

### CODE REVIEW

File: clang/lib/Sema/SemaOpenMP.cpp [hunk 1]
```diff
@@ -3010,6 +3010,28 @@ static bool FinishOpenMPLinearClause(OMPLinearClause &Clause, DeclRefExpr *IV,
...
+                                DSAStackTy *Stack) {
+  assert((SemaRef.CurContext->isDependentContext() || B.builtAll()) &&
+         "loop exprs were not built");
+
+  if (SemaRef.CurContext->isDependentContext())
+    return false;
+
+  // Finalize the clauses that need pre-built expressions for CodeGen.
+  for (OMPClause *C : Clauses) {
+    if (auto *LC = dyn_cast<OMPLinearClause>(C)) {
```
Comment (alexey-bataev) [inline clang/lib/Sema/SemaOpenMP.cpp:3024]:
```
auto *LC = dyn_cast<OMPLinearClause>(C);
if (!LC)
  continue;
```

File: clang/lib/Sema/SemaOpenMP.cpp [hunk 2]
```diff
@@ -7994,13 +8016,9 @@ void SemaOpenMP::ActOnOpenMPDeclareVariantDirective(
...
-SemaOpenMP::ActOnOpenMPParallelDirective(ArrayRef<OMPClause *> Clauses,
-                                         Stmt *AStmt, SourceLocation StartLoc,
-                                         SourceLocation EndLoc) {
-  if (!AStmt)
-    return StmtError();
-
+static CapturedStmt *
+setBranchProtectedScope(Sema &SemaRef, OpenMPDirectiveKind DKind, Stmt *AStmt) {
+  assert(isa<CapturedStmt>(AStmt) && "Captured statement expected");
   auto *CS = cast<CapturedStmt>(AStmt);
```
Comment (alexey-bataev) [inline clang/lib/Sema/SemaOpenMP.cpp:8022]:
```
auto *CS = dyn_cast<CapturedStmt>(AStmt);
assert(CS && ...);
```

File: clang/lib/Sema/SemaOpenMP.cpp (hunk 1 above)
Comment (kparzysz) [reply clang/lib/Sema/SemaOpenMP.cpp]: Done.

**Review by alexey-bataev — APPROVED** [review]
LG

**Pull request merged**